src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -181,14 +246,32 @@ def download(self, file_name: str, before_downloaded: Optional[Callable] = None)
         before_downloaded: Callable
             Write a file.
         """
-        if before_downloaded:
-            before_downloaded(
-                os.path.basename(file_name) if pypim.is_configured() else file_name
-            )
-        if pypim.is_configured():
-            if os.path.isfile(file_name):
-                print(f"\nFile already exists. File path:\n{file_name}\n")
-            else:
-                self._transfer_service.download(
-                    os.path.basename(file_name), local_file_name="."
-                )
+        self._transfer_service.download_file(
+            file_name=file_name, before_downloaded=before_downloaded
+        )
+
+    def __bool__(self):
+        if self._service_type == ServiceType.PIM:
+            return self._transfer_service.is_pim_configured()
+        if self._service_type == ServiceType.Test:
+            return False
+
+
+class MyFileService:
```

Comment:
@hpohekar You probably ought to call it MockFileService (for instance). I used this name to make the general point.

Reply:
Yes, Thank you.